File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/HostClusterAvailabilityCheckerTask.java
package com.sequenceiq.cloudbreak.core.bootstrap.service.host;

import com.sequenceiq.cloudbreak.core.bootstrap.service.host.context.HostOrchestratorClusterContext;
import com.sequenceiq.cloudbreak.service.StackBasedStatusCheckerTask;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class HostClusterAvailabilityCheckerTask extends StackBasedStatusCheckerTask<HostOrchestratorClusterContext> {

    private static final Logger LOGGER = LoggerFactory.getLogger(HostClusterAvailabilityCheckerTask.class);

    @Override
    public boolean checkStatus(HostOrchestratorClusterContext context) {
        List<String> missingNodes = context.getHostOrchestrator().getMissingNodes(context.getGatewayConfig(), context.getNodes());
        LOGGER.debug("Missing nodes from orchestrator cluster: {}", missingNodes);
        return missingNodes.isEmpty();
    }

    @Override
    public void handleTimeout(HostOrchestratorClusterContext t) {
    }

    @Override
    public String successMessage(HostOrchestratorClusterContext t) {
        return "Host orchestration API is available and the agents are registered.";
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/HostClusterAvailabilityCheckerTask.kt
package com.sequenceiq.cloudbreak.core.bootstrap.service.host

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.core.bootstrap.service.host.context.HostOrchestratorClusterContext
import com.sequenceiq.cloudbreak.service.StackBasedStatusCheckerTask

@Component
class HostClusterAvailabilityCheckerTask : StackBasedStatusCheckerTask<HostOrchestratorClusterContext>() {

    override fun checkStatus(context: HostOrchestratorClusterContext): Boolean {
        val missingNodes = context.hostOrchestrator.getMissingNodes(context.gatewayConfig, context.nodes)
        LOGGER.debug("Missing nodes from orchestrator cluster: {}", missingNodes)
        return missingNodes.isEmpty()
    }

    override fun handleTimeout(t: HostOrchestratorClusterContext) {
        return
    }

    override fun successMessage(t: HostOrchestratorClusterContext): String {
        return "Host orchestration API is available and the agents are registered."
    }

    companion object {

        private val LOGGER = LoggerFactory.getLogger(HostClusterAvailabilityCheckerTask::class.java)
    }
}
